vllm/entrypoints/api_server.py

```diff
@@ -43,15 +44,21 @@ async def generate(request: Request) -> Response:
                                         request_id,
                                         prefix_pos=prefix_pos)
 
+    label = {"endpoint": "/generate"}
+
     # Streaming case
     async def stream_results() -> AsyncGenerator[bytes, None]:
         async for request_output in results_generator:
-            prompt = request_output.prompt
-            text_outputs = [
-                prompt + output.text for output in request_output.outputs
-            ]
-            ret = {"text": text_outputs}
-            yield (json.dumps(ret) + "\0").encode("utf-8")
+            if request_output.finished:
+                counter_inference_request_success.inc(label)
+                prompt = request_output.prompt
+                text_outputs = [
+                    prompt + output.text for output in request_output.outputs
+                ]
+                ret = {"text": text_outputs}
+                yield (json.dumps(ret) + "\0").encode("utf-8")
```

Comment:
wasn't this already running in the aborted case? are we intentionally moving it inside the if statement?